packages/cdktf/lib/terraform-dynamic-block.ts

```diff
@@ -0,0 +1,71 @@
+import { TerraformDynamicExpression } from "./terraform-dynamic-expression";
+import { ITerraformIterator } from "./terraform-iterator";
+import { IResolvable, IResolveContext, Lazy, Token } from "./tokens";
+import { captureStackTrace } from "./tokens/private/stack-trace";
+
+const DYNAMIC_BLOCK_SYMBOL = Symbol.for("cdktf/TerraformDynamicBlock");
+
+export class TerraformDynamicBlock implements IResolvable {
+  public readonly creationStack: string[];
+  public readonly forEach: ITerraformIterator;
+  public readonly iterator = "each"; // name of temporary variable used in iteration, needed in the future when allowing nesting iterators
+  public readonly content: { [key: string]: any };
+  // labels: TODO: support labels, but they seem to be quite rare (see https://discuss.hashicorp.com/t/labels-in-dynamic-block/21461/5)
```

Comment:
We should make a follow up issue to support labels

Reply:
done